project/dexter-vs/Dexter/Analyzer/Core/NoCommentAnalyzer.cs

```diff
@@ -0,0 +1,331 @@
+                .Where(attribute => IsNameAttribute(attribute, "exception"))
+                .Select(attribute => attribute.Identifier.ToString());
+
+            var exceptionTokens = GetExceptionTokens(node);
+
+            foreach (var exceptionToken in exceptionTokens)
+            {
+                var exceptionName = exceptionToken.Text;
+
+                if (!exceptionCommentNames.Any(commentName => commentName.Equals(exceptionName)))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(NoExceptionRule, exceptionToken.GetLocation(), exceptionToken.Text));
+                }
+            }
+        }
+
+        private static IEnumerable<SyntaxToken> GetExceptionTokens(SyntaxNode node)
+        {
+            var throwStatmentSyntaxs = node.DescendantNodes().OfType<ThrowStatementSyntax>();
+            var ObjectCreationExpressions = from throwStatment in throwStatmentSyntaxs
+                                            where throwStatment.Expression != null &&
+                                            throwStatment.Expression is ObjectCreationExpressionSyntax
+                                            select throwStatment.Expression as ObjectCreationExpressionSyntax;
+
+            var exceptionTokens = from expression in ObjectCreationExpressions
+                                  let type = expression.Type as IdentifierNameSyntax
+                                  where type != null
+                                  select type.Identifier;
+
+            return exceptionTokens;
+        }
+
+        private void ReportNoParamDiagnostics(SyntaxNodeAnalysisContext context, BaseMethodDeclarationSyntax node, DocumentationCommentTriviaSyntax xmlTrivia)
+        {
+            var paramCommentNames = xmlTrivia.DescendantNodes().OfType<XmlNameAttributeSyntax>()
+                .Where(attribute => IsNameAttribute(attribute, "param"))
+                .Select(attribute => attribute.Identifier.ToString());
+
+            foreach (var parameter in node.ParameterList.Parameters)
+            {
+                var paramName = parameter.Identifier.Text;
+
+                if (HasThisModifier(parameter))
+                    continue;
+
+                if (!paramCommentNames.Any(commentName => commentName.Equals(paramName)))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(NoParamRule, parameter.Identifier.GetLocation(), parameter.Identifier.Text));
+                }
+            }
+        }
+
+        private bool HasThisModifier(ParameterSyntax parameter)
+        {
+            return parameter.Modifiers.Any(modifier => modifier.Text.Equals("this"));
+        }
+
+        private bool IsNameAttribute(XmlNameAttributeSyntax xmlNameAttribute, string name)
+        {
+            var parent = xmlNameAttribute.Parent as XmlElementStartTagSyntax;
+            if (parent == null)
```

Comment:
Even though statement are a single line, brace should be used for If Statement. It is VD C# CRC, please use to the next commit. :-)